packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -717,7 +717,10 @@ export const Table = <
   // This is to avoid the old columnOrder from affecting the new columns' columnOrder
   React.useEffect(() => {
     // Check if columns have changed (by value)
-    if (JSON.stringify(lastPassedColumns.current) !== JSON.stringify(columns)) {
+    if (
+      lastPassedColumns.current.length > 0 &&
+      JSON.stringify(lastPassedColumns.current) !== JSON.stringify(columns)
+    ) {
       instance.setColumnOrder([]);
     }
     lastPassedColumns.current = columns;
```

Comment:
Some comments for future:

1. We should try to avoid stringifying columns like this. It can be expensive and error-prone. In our _new_ Table we could require always passing an id, or find a way to compare the internal ids.
 
2. We should maybe create a hook for this "previous state" pattern.
   ```jsx
   function usePreviousState<T>(state: T, onChange: () => void): React.Ref<T>;
   ```

Reply:
Sounds good, linked this comment in https://github.com/iTwin/iTwinUI/issues/1144.